### PR TITLE
Telemetry: new API, add bundle add/remove telemetry.

### PR DESCRIPTION
**Pull request by sofar**
- Change telemetry API - add class component. 
- Telemetry: add telemetry for bundle add, remove.

### CODE REVIEW

File: src/bundle.c [hunk 1]
```diff
@@ -552,13 +560,29 @@ int install_bundles_frontend(char **bundles)
 
 	for (; *bundles; ++bundles) {
 		bundles_list = list_prepend_data(bundles_list, *bundles);
+		if (*bundles) {
+			if (bundles_list_str) {
+				string_or_die(&bundles_list_str, "%s, %s", bundles_list_str, *bundles);
```
Comment (phmccarty) [inline src/bundle.c]: You will want to `free(bundles_list_str)` before this line to avoid a memory leak.

Comment (phmccarty) [inline src/bundle.c]: Or rather, store the old `bundles_list_str` in a temp variable before freeing, and use the temp variable in the `string_or_die()`.

Comment (ahkok) [inline src/bundle.c]: Ok, I'll fix that.

**Comment by ahkok** [comment]
Updated to fix a minor leak.

**Comment by phmccarty** [comment]
+1

**Pull request merged**